src/components/forms/ClusteredDiskSizeSelector.tsx

```diff
@@ -0,0 +1,231 @@
+import { FC, Fragment, useEffect, useState } from "react";
+import {
+  CheckboxInput,
+  Input,
+  Label,
+  Select,
+} from "@canonical/react-components";
+import ResourceLink from "components/ResourceLink";
+import FormEditButton from "components/FormEditButton";
+import { ClusterSpecificValues } from "components/ClusterSpecificSelect";
+import { useClusterMembers } from "context/useClusterMembers";
+import { parseMemoryLimit } from "util/limits";
+import { BYTES_UNITS } from "types/limits";
+interface Props {
+  id: string;
+  toggleReadOnly: () => void;
+  setMemoryLimit: (value: ClusterSpecificValues) => void;
+  isReadOnly?: boolean;
+  sizeValues?: ClusterSpecificValues;
+  canToggleSpecific?: boolean;
+  isDefaultSpecific?: boolean;
+  clusterMemberLinkTarget?: (member: string) => string;
+  disabled?: boolean;
+  helpText?: string;
+}
+
+const ClusteredDiskSizeSelector: FC<Props> = ({
+  id,
+  isReadOnly = false,
+  setMemoryLimit,
+  sizeValues,
+  toggleReadOnly,
+  canToggleSpecific = true,
+  isDefaultSpecific = null,
+  clusterMemberLinkTarget = () => "/ui/cluster",
+  disabled = false,
+  helpText,
+}) => {
+  const { data: clusterMembers = [] } = useClusterMembers();
+  const memberNames = clusterMembers.map((member) => member.server_name);
+
+  const [isSpecific, setIsSpecific] = useState<boolean | null>(
+    isDefaultSpecific,
+  );
+  const firstValue = Object.values(sizeValues ?? {})[0];
+
+  const limit = (val: string) => {
```

Comment:
```suggestion
  const parseLimit = (val: string) => {
```
That should be easier to read.